backend/src/test/java/com/google/sps/servlets/AccountServletTest.java

```diff
@@ -0,0 +1,163 @@
+  private static final String HOME_LINK = "/home";
+
+  private static final LocalServiceTestHelper DATASTORE_HELPER = TestUtils.getDatastoreHelper();
+
+  private static DatastoreService datastore;
+
+  @Mock HttpServletRequest request;
+  @Mock HttpServletResponse response;
+  @Mock UserService userService;
+
+  @Before
+  public void setUp() {
+    DATASTORE_HELPER.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    MockitoAnnotations.openMocks(this);
+    when(userService.isUserLoggedIn()).thenReturn(true);
+    when(userService.getCurrentUser()).thenReturn(TestUtils.USER);
+  }
+
+  @After
+  public void tearDown() {
+    DATASTORE_HELPER.tearDown();
+  }
+
+  private AccountServlet getAccountServlet() {
+    AccountServlet servlet = new AccountServlet();
+    servlet.setUserServiceForTesting(userService);
+    return servlet;
+  }
+
+  private void getReturnsUserDetails(Entity user, UserInfo expectedUser) throws Exception {
+    datastore.put(user);
+    assertEquals(1, datastore.prepare(new Query(UserConstants.ENTITY_USER)).countEntities(withLimit(10)));
+
+    String result =
+        TestUtils.getResultFromAuthenticatedGetRequest(getAccountServlet(), request, response);
+    assertEquals(GSON.toJson(expectedUser), result);
+  }
+
+  @Test
+  public void getRedirectsToSignUpIfNoName() throws Exception {
+    Entity user = TestUtils.getEmptyUserEntity();
+    datastore.put(user);
+    assertEquals(1, datastore.prepare(new Query(UserConstants.ENTITY_USER)).countEntities(withLimit(10)));
+
+    getAccountServlet().doGet(request, response);
+    verify(response).sendRedirect(AccountServlet.SIGN_UP_LINK);
+  }
+
+  @Test
+  public void getReturnsUserDetailsWithDietsAndAllergies() throws Exception {
+    Entity user = TestUtils.getUserEntityWithDietsAndAllergies();
+    UserInfo expectedUser = new UserInfo(TestUtils.NAME, TestUtils.DIETS, TestUtils.ALLERGIES);
+    getReturnsUserDetails(user, expectedUser);
+  }
+
+  @Test
+  public void getReturnsUserDetailsWithDietsOnly() throws Exception {
+    Entity user = TestUtils.getUserEntityWithDietsOnly();
+    UserInfo expectedUser = new UserInfo(TestUtils.NAME, TestUtils.DIETS, Collections.emptyList());
+    getReturnsUserDetails(user, expectedUser);
+  }
+
+  @Test
+  public void getReturnsUserDetailsWithAllergiesOnly() throws Exception {
+    Entity user = TestUtils.getUserEntityWithAllergiesOnly();
+    UserInfo expectedUser =
+        new UserInfo(TestUtils.NAME, Collections.emptyList(), TestUtils.ALLERGIES);
+    getReturnsUserDetails(user, expectedUser);
+  }
+
+  @Test
+  public void getReturnsUserDetailsWithNoDietsAndAllergies() throws Exception {
+    Entity user = TestUtils.getUserEntityWithName();
+    UserInfo expectedUser =
+        new UserInfo(TestUtils.NAME, Collections.emptyList(), Collections.emptyList());
+    getReturnsUserDetails(user, expectedUser);
+  }
+
+  @Test
+  public void postCreatesNewUserAndAddsToDatastore() throws Exception {
```

Comment:
perhaps in a followup, consider adding malformed post requests to verify expected behavior.